quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -59,13 +62,17 @@ public void beforeEach(ExtensionContext context) throws Exception {
                 if (arg.contains(CONFIG_FILE_SHORT_NAME) || arg.contains(CONFIG_FILE_LONG_NAME)) {
                     Pattern kvSeparator = Pattern.compile(KEY_VALUE_SEPARATOR);
                     String[] cfKeyValue = kvSeparator.split(arg);
-                    System.setProperty(KeycloakPropertiesConfigSource.KEYCLOAK_CONFIG_FILE_PROP, cfKeyValue[1]);
+                    setProperty(KeycloakPropertiesConfigSource.KEYCLOAK_CONFIG_FILE_PROP, cfKeyValue[1]);
+                } else if (distConfig == null && arg.startsWith("-D")) {
+                    // allow setting system properties from JVM tests
+                    String name = arg.substring(2, arg.indexOf('='));
```

Comment:
this implementation handle only properties using using the `=` sign.
Probably we can have a more robust implementation using something like `new Properties().load(new StringReader(s))` and only stripping the trailing `-D`

Reply:
How the robust implementation would look like?

Reply:
I was thinking about something on those lines:

```java
System.setProperties(new Properties().load(new StringReader(s.replaceFirst("-D", ""))));
```

Reply:
I'm a bit lost, sorry. This is supposed to mimic passing system props (JVM) when running JVM tests. What other properties do you want to enable that do not use the `=` sign?

Your proposal is also replacing all system properties, we just want to add new properties.

We can probably add a check to make sure the code does not break if the CLI arg does not hold a `=`. But I do not see why the good and old substring is not enough here.

Reply:
is a nitpick, no need to continue this discussion any further, not fussy on it, resolved.